nflow-rest-api-jax-rs/src/main/java/io/nflow/rest/v1/jaxrs/WorkflowInstanceResource.java

```diff
@@ -147,4 +151,16 @@ public Response setSignal(@ApiParam("Internal id for workflow instance") @PathPa
     return (updated ? ok("Signal was set successfully") : ok("Signal was not set")).build();
   }
 
+  @PUT
+  @Path("/{id}/wakeup")
+  @ApiOperation(value = "Wake up sleeping workflow instance. If expected states are given, only wake up if the instance is in one of the expected states.")
+  @ApiResponses({ @ApiResponse(code = 200, message = "When workflow wakeup was attempted")})
+  public WakeupResponse wakeup(@ApiParam("Internal id for workflow instance") @PathParam("id") int id,
+      @Valid @ApiParam("Expected states") WakeupRequest req) {
+    WakeupResponse response = new WakeupResponse();
+    List<String> expectedStates = req.expectedStates != null ? req.expectedStates : emptyList();
```

Comment:
```suggestion
    List<String> expectedStates = Optional.ofNullable(req.expectedStates).orElseGet(Collections::emptyList);
```

Reply:
Fixed. Not sure though if it is better...